commands.txt

```diff
@@ -23,3 +23,5 @@ $SD_CURL_CMD_WRAPPER "curl -o ./sonarscanner.zip -L https://binaries.sonarsource
 $SD_UNZIP_CMD_WRAPPER "unzip -q ./sonarscanner.zip"
 SCANNER_PATH="sonar-scanner-${SD_SONAR_SCANNER_CLI_VERSION}-${TARGET_OS}/bin/sonar-scanner"
 eval $SCANNER_PATH $SCANNER_OPTION
+if [ ! -z "$SD_SELF_SONAR_HOST" ]; then SD_SONAR_PROJECT_URL='${SD_SELF_SONAR_HOST}/dashboard?id=${$SD_SONAR_PROJECT_KEY}'; else SD_SONAR_PROJECT_URL='${SD_SONAR_HOST}/dashboard?id=${$SD_SONAR_PROJECT_KEY}'; fi
```

Comment:
we are making an API call at line 5. I would prefer we handle it in a single call.

Reply:
They are calling different endpoints though. 

I moved the logic at the end to make sure the sonar project is created before setting.

Reply:
![Screenshot 2023-10-31 at 2 11 32 PM](https://github.com/screwdriver-cd/coverage-sonar/assets/15989893/5053495a-

```
SONAR_RESPONSE='{"token": "somekey","projectUrl": "https://sonar.sd.cd/dashboard=123"}'
SONAR_TOKEN="$(jq -jn "$SONAR_RESPONSE|.token")"
SD_SONAR_PROJECT_URL="$(j
0cf1-4fe7-9858-1a9ab6f8ce3b)
q -jn "$SONAR_RESPONSE|.projectUrl")"
echo $SONAR_TOKEN
echo $SD_SONAR_PROJECT_URL
```